### ISSUE TITLE
[ICON BUG CONTEXT LOG] CSS overrides of `Icon` `fill` may need more specificity going forward

### ISSUE THREAD

**Issue by chloerice** [Shopify/polaris-react-archive#11358]
## Tl;dr: 

**If you have a custom component that overrides the `fill` of an SVG rendered by the Polaris Icon component, you may need to fix forward by either setting `tone="inherit"` on the `Icon` a/o by updating your custom styles to target `color` instead of `fill`.**

Major changes have recently shipped to icons across:
- Shopify core web infrastructure ([removal ](https://github.com/Shopify/web/pull/110790)of the `replaceAttrValues` script)
- `@shopify/polaris-icons` (removal of hard coded `fill`)
- `Icon` `svg` `fill` styles now inherit from `color` via `currentColor`
 
This was not intended to be a breaking change and very few issues have been raised so far, however it's important this context is shared before the EOY shutdown as many folks will be OOO through the first week of January. 

## Example 

"View in online store" and "View template" icon only custom links in the `/products` section:

| /products| /products/:id |
|--------|--------|
| ![image (1)](https://github.com/Shopify/web/assets/18447883/849c4918-6ad4-4cef-8cff-a1eee05b35a5) | <img width="1890" alt="image (2)" src="https://github.com/Shopify/web/assets/18447883/596c65ac-83fa-4b30-8b1a-60ddf8577fff"> | 

## Context on icon use, customization, and build changes

(Thank you @alex-page 🙏🏽 )

**Previous state -- <=12.2.0**

- <Icon> component did nothing to the `svg` `fill` or `color` by default
- `Icon` `tone` did not have a default value, and fell back to the browser default of `rgb(0, 0, 0,)` (black)
- Changing the `tone` sets the `fill` value
- `@shopify/polaris-icons` had a `fill` value that was hard coded, removed on build
- Custom icons could be any color they wanted to be

**Now -- >= 12.3.0**

- `<Icon>` component does nothing to the `svg` `fill` or `color` by default
- `Icon` `fill` can inherit from parent `color` declarations (including browser defaults no overridden by parent tree)
- `Icon` `tone` does not have a default value applied
- Setting `tone` overrides the `color` property
- polaris-icons does not have a fill value. This is now matches what our old build script did.
- Custom icons can be any color they want to be

**Potential approach [explored](https://github.com/Shopify/polaris/pull/11355) with default icon value of  `--p-color-icon`**

- `<Icon>` component sets a default color
- `Icon` `tone` has a default value
- Changing the tone changes the color value
- `@shopify/polaris-icons` does not have a `fill` value. This is now matches what our old build script did.
- Custom icons need more specificity to override the default value. If they have a fill or color set on the SVG or in the custom code it could lose to the default value from specificity.

The challenge with _Now_ is that some folks used `<Icon>` with no `tone`and added CSS overrides to the component's parent elements reaching into `svg` or `path` to modify the fill value. This causes odd problems very rarely as before it would go up the tree and find a `fill` and now it is going up the tree to find a color. In these scenarios we can update the override code to be color or replace the `<Icon>` `tone` with `inherit`. The [product index is a good example of that](https://github.com/Shopify/web/blob/999155390922f28d6efbbb2254c5b007f50edac5/app/sections/Products/ProductIndex/components/ProductIndexTable/components/ProductIndexTableRow/components/TitleCell/TitleCell.css#L49-L72), where they override for specific values but never set a base color and likely rendered a black icon until now.

**Comment by alex-page** [comment]
IMO this issue can be closed and we can continue to fix forward any previous hacky icon overrides.
